main.tf

```diff
@@ -152,3 +152,22 @@ module "vm" {
     module.configuration
   ]
 }
+
+
+module "dns" {
+  source               = "./modules/dns"
+  resource_name_prefix = var.resource_name_prefix
+  resource_group_name  = var.resource_group_name
+  zone_dns_name        = var.zone_dns_name
+  # principal_id         = module.identity.identity_name
+  # virtual_network_id   = azurerm_virtual_network.graphdb.id
+  identity_name = module.identity.identity_name
+  depends_on = [
```

Comment:
We should discuss the use of `depends_on` + data sources. HashiCorn [discourage](https://developer.hashicorp.com/terraform/language/meta-arguments/depends_on) the use of `depends_on`, unless absolutely necessary and it seems that the use of data sources should be for things that are [assumed to exist](https://support.hashicorp.com/hc/en-us/articles/4414363057299-Best-Practice-depends-on-Meta-Argument).